src/cgame/cg_rocket_draw.cpp

```diff
@@ -2174,6 +2174,133 @@ class PredictedMineEfficiencyElement : public HudElement
 	std::unordered_map<int, std::string> pluralSuffix;
 };
 
+class BarbsHudElement : public HudElement
+{
+public:
+	BarbsHudElement ( const Rocket::Core::String& tag ) :
+	HudElement ( tag, ELEMENT_ALIENS ),
+	numBarbs( 0 ),
+	regenerationInterval ( 0 ),
+	t0 ( 0 ),
+	offset ( 0 ) {}
+
+	void OnAttributeChange( const Rocket::Core::AttributeNameList& changed_attributes )
+	{
+		HudElement::OnAttributeChange( changed_attributes );
+		if ( changed_attributes.find( "src" ) != changed_attributes.end() )
+		{
+			int m = BG_Weapon( WP_ALEVEL3_UPG )->maxAmmo;
+			if ( m > 0 )
+			{
+				char base[ MAX_STRING_CHARS ];
```

Comment:
No need to use char. You can use Rocket::Core::String::Format(<length>, "format", ...);
str += text to append